### PR TITLE
Added templates for azure pipelines

### PR DESCRIPTION
**Pull request by bosesuneha**
# Description
Added templates for azure pipelines, to be used in DevHub ADO integration 

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: template/azurePipelines/kustomize/draft.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - name: "CLUSTER_RESOURCE_GROUP"
+    description: "the AKS cluster resource group"
+  - name: "ACR_RESOURCE_GROUP"
+    description: "the ACR resource group"
+  - name: "CLUSTER_NAME"
+    description: "the AKS cluster name"
+  - name: "KUSTOMIZE_PATH"
+    default:
+      disablePrompt: true
+      value: "./overlays/production"
```
Comment (Tatsinnit) [inline template/azurePipelines/kustomize/draft.yaml:25]: 💡 Quick note: This value is some sort of preferred default value, (if documented anywhere can we comment add the reason please)? WDYT?

**Review by Tatsinnit — APPROVED** [review]
☕️❤️ Thank you so much for the wider ping, looks good to me, mostly the scaffolding infra I guess. If you have reasonable docs around this, feel free to add as comment will be only minor thing. rest looks good to me. Thank you!! ☕️🙏

File: template/azurePipelines/kustomize/draft.yaml (hunk 1 above)
Comment (bosesuneha) [reply template/azurePipelines/kustomize/draft.yaml]: Thanks for the suggestion, I've added a comment beside it.

File: template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Azure Kubernetes Service (AKS) pipeline with Kustomize
+# Build and push image to Azure Container Registry; Deploy to Azure Kubernetes Service cluster
+
+variables:
+  armServiceConnection: {{ARM_SERVICE_CONNECTION}}
```
Comment (OliverMKing) [inline template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml:5]: this seems like a weird name for what's the azure subscription

File: template/azurePipelines/kustomize/draft.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+variables:
+  - name: "PIPELINE_NAME"
+    default:
+      value: "Build and deploy an app to AKS"
+    description: "the name of the azure pipeline"
+  - name: "BRANCH_NAME"
+    default:
+      value: "main"
+    description: "the branch to trigger the pipeline"
+  - name: "ARM_SERVICE_CONNECTION"
```
Comment (OliverMKing) [inline template/azurePipelines/kustomize/draft.yaml:10]: this is really just the azure subscription right? Renaming might be good

File: template/azurePipelines/manifests/.pipelines/azure-kubernetes-service.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  clusterName: {{CLUSTER_NAME}}
+  manifestPath: {{MANIFEST_PATH}}
+  namespace: {{NAMESPACE}}
+  tag: "$(Build.BuildId)"
+  vmImageName: "ubuntu-latest"
+
+name: {{PIPELINE_NAME}}
+
+trigger:
+  - main
```
Comment (OliverMKing) [inline template/azurePipelines/manifests/.pipelines/azure-kubernetes-service.yaml:19]: trigger is hardcoded here when it's not in kustomize

File: template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml (hunk 2 above)
Comment (bosesuneha) [reply template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml:5]: It's the name of the Azure Resource Manager service connection, not the subscription

Comment (OliverMKing) [reply template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml:5]: The docs and how it's used here seem to imply it's the subscription. What is the ARM Service Connection?

File: template/azurePipelines/manifests/.pipelines/azure-kubernetes-service.yaml (hunk 4 above)
Comment (bosesuneha) [reply template/azurePipelines/manifests/.pipelines/azure-kubernetes-service.yaml]: Thanks for catching that, changed it into a variable

File: template/azurePipelines/kustomize/draft.yaml (hunk 3 above)
Comment (bosesuneha) [reply template/azurePipelines/kustomize/draft.yaml:10]: No, it's the ADO service connection that is linked to an msi to access the AKS cluster and ACR

File: template/azurePipelines/kustomize/draft.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+variables:
+  - name: "PIPELINE_NAME"
```
Comment (Vidya2606) [inline template/azurePipelines/kustomize/draft.yaml:2]: to keep it consistent, let's not use the separators unless you have a specific reason. Can just keep it "PIPELINENAME"
https://github.com/Azure/draft/blob/main/template/workflows/helm/draft.yaml

File: template/azurePipelines/manifests/draft.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+variables:
+  - name: "PIPELINE_NAME"
```
Comment (Vidya2606) [inline template/azurePipelines/manifests/draft.yml:2]: same here, we can remove separator

File: template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml (hunk 2 above)
Comment (bosesuneha) [reply template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml:5]: https://learn.microsoft.com/en-us/azure/devops/pipelines/library/connect-to-azure?view=azure-devops#create-service-connection-with-workload-identity-federation-manual

This is what we'll be using.
I know it says Azure subscription in the [KubernetesManifest@1](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/kubernetes-manifest-v1?view=azure-pipelines) doc, but it does not take the subscription ID or name and expects the service connection name.

**Review by OliverMKing — APPROVED** [review]

File: template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml (hunk 2 above)
Comment (OliverMKing) [reply template/azurePipelines/kustomize/.pipelines/azure-kubernetes-service-kustomize.yaml]: Makes sense to me!

File: template/azurePipelines/kustomize/draft.yaml (hunk 5 above)
Comment (bosesuneha) [reply template/azurePipelines/kustomize/draft.yaml]: removed the separators to maintain consistency with the other templates

File: template/azurePipelines/manifests/draft.yml (hunk 6 above)
Comment (bosesuneha) [reply template/azurePipelines/manifests/draft.yml]: I've removed the separators from the var names

**Pull request merged**